### DISCUSSION TITLE
Next/Nuxt support?

### DISCUSSION THREAD

**Discussion by OzzyTheGiant** [TypedProject/ts-express-decorators#1543] [Q&A]
Just wondering, is it possible to integrate this into a Next or Nuxt project given that they all use Express as their back end?

**Comment by Romakita** [comment]
@OzzyTheGiant I don't use Next / Nuxt but I can help you to integrate Ts.ED with next / nuxt.

**Reply by OzzyTheGiant** [reply]
Thanks, but if you meant hands-on help, that won't be necessary. I'm asking for a private project being worked on by multiple teams. Product Owners locked in TS.ED as our back end framework, but there is currently a discussion about whether this is necessary given that Next already has its own back end. So I just wanted to know a brief overview of how I would go about doing that.

**Reply by Romakita** [reply]
I'm sorry if I didn't understand the first request, but you wished you had support on Ts.ED and next / nuxt ^^

>Just wondering, is it possible to integrate this into a Next or Nuxt project given that they all use Express as their back end?

So yes it's probably possible. Ts.ED accept an external expressApp to load the routes.

>Thanks, but if you meant hands-on help, that won't be necessary. I'm asking for a private project being worked on by multiple teams. Product Owners locked in TS.ED as our back end framework, but there is currently a discussion about whether this is necessary given that Next already has its own back end. So I just wanted to know a brief overview of how I would go about doing that.

Next and Ts.ED doesn't propose the same feature... Next is design for SSR but it don't provide the features about the Rest API provided by Ts.ED (like JsonSchema, Models, DI, etc...). I haven't in mind all possibility of the backend proposed by Next, but it's pretty close to what we would write in pure express.js, right ?

**Reply by OzzyTheGiant** [reply]
Ok I think that somewhat answers what I'm looking for. Yes, Next is about SSR, although it provides a way to expose the express app for configuration. I guess the question all along was if I can attach TS.ED controllers to Next's app object. I think maybe with some tinkering this would be possible.

**Reply by Romakita** [reply]
When you bootstrap a Ts.ED application, it's possible to give the expressApp for another application like this:

```typescript
const expressApp = express()


PlatformExpress.boostrap(Server, {
   express: {
        app: expressApp
      }
})
```
